travis failure in p2p-segwit.py not(b'bad-txns-vout-negative' == b'bad-witness-nonstandard')

This keeps popping up from time to time on travis. The relevant log excerpt:

```
2017-12-19 17:52:57.940000 TestFramework (INFO): Testing detection of non-standard P2WSH witness
2017-12-19 17:52:58.710000 TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "./test/functional/test_framework/test_framework.py", line 118, in main
    self.run_test()
  File "./test/functional/p2p-segwit.py", line 1943, in run_test
    self.test_non_standard_witness()
  File "./test/functional/p2p-segwit.py", line 1827, in test_non_standard_witness
    test_transaction_acceptance(self.nodes[1].rpc, self.std_node, p2wsh_txs[0], True, False, b'bad-witness-nonstandard')
  File "./test/functional/p2p-segwit.py", line 48, in test_transaction_acceptance
    assert_equal(p2p.last_message["reject"].reason, reason)
  File "./test/functional/test_framework/util.py", line 38, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(b'bad-txns-vout-negative' == b'bad-witness-nonstandard')
```


Most recent example: https://travis-ci.org/bitcoin/bitcoin/jobs/318711688

--- a/test/functional/p2p-segwit.py
+++ b/test/functional/p2p-segwit.py
@@ -1396,18 +1396,20 @@ class SegWitTest(BitcoinTestFramework):
 
         temp_utxos.pop(0)
 
-        # Update self.utxos for later tests. Just spend everything in
-        # temp_utxos to a corresponding entry in self.utxos
+        # Update self.utxos for later tests by creating two outputs
+        # that consolidate all the coins in temp_utxos.
+        output_value = sum(i.nValue for i in temp_utxos) // 2
+
         tx = CTransaction()
         index = 0
+        # Just spend to our usual anyone-can-spend output
+        tx.vout = [CTxOut(output_value, CScript([OP_TRUE]))] * 2
         for i in temp_utxos:
-            # Just spend to our usual anyone-can-spend output
-            # Use SIGHASH_SINGLE|SIGHASH_ANYONECANPAY so we can build up
+            # Use SIGHASH_ALL|SIGHASH_ANYONECANPAY so we can build up
             # the signatures as we go.
             tx.vin.append(CTxIn(COutPoint(i.sha256, i.n), b""))
-            tx.vout.append(CTxOut(i.nValue, CScript([OP_TRUE])))
             tx.wit.vtxinwit.append(CTxInWitness())
-            sign_P2PK_witness_input(witness_program, tx, index, SIGHASH_SINGLE|SIGHASH_ANYONECANPAY, i.nValue, key)
+            sign_P2PK_witness_input(witness_program, tx, index, SIGHASH_ALL|SIGHASH_ANYONECANPAY, i.nValue, key)
             index += 1
         block = self.build_next_block()
         self.update_witness_block_with_transactions(block, [tx])

